packages/manager/apps/container/src/container/nav-reshuffle/sidebar/index.tsx

```diff
@@ -163,41 +176,49 @@ const Sidebar = (): JSX.Element => {
   };
 
   useEffect(() => {
-    if (selectedNode) return;
-
     const pathname = location.pathname;
-    const savedNodeID = window.localStorage.getItem(savedLocationKey);
-    if (!savedNodeID) return;
-    const node = findNodeById(currentNavigationNode, savedNodeID);
-    if (!node || !node.universe) {
-      window.localStorage.removeItem(savedLocationKey);
-      return;
+    let savedNode: Node = null;
+    if (currentNavigationNode.id !== 'sidebar') return;
+    if (savedNodeID && !selectedSubMenu) {
+      savedNode = findNodeById(currentNavigationNode, savedNodeID);
+      if (!savedNode || !savedNode.universe) {
+        setSavedNode(null);
+        return;
+      }
     }
 
-    const parent = findNodeById(currentNavigationNode, node.universe);
-    if (parent) {
-      const nodePath = node.routing.hash
-        ? node.routing.hash.replace('#', node.routing.application)
-        : '/' + node.routing.application;
-
-      const parsedPath = splitPathIntoSegmentsWithoutRouteParams(nodePath);
-      const isMatching = parsedPath.reduce(
-        (acc: boolean, segment: string) => acc && pathname.includes(segment),
-        true,
-      );
-      if (isMatching) {
-        selectSubMenu(node, parent);
-        return;
-      } else {
-        window.localStorage.removeItem(savedLocationKey);
+    const currentNode: Node = selectedSubMenu || savedNode;
+    if (currentNode) {
+      const parent = findNodeById(currentNavigationNode, currentNode.universe);
+      if (parent) {
+        const currentNodePath = currentNode.routing.hash
+          ? currentNode.routing.hash.replace(
+              '#',
+              currentNode.routing.application,
+            )
+          : '/' + currentNode.routing.application;
+
+        const parsedPath = splitPathIntoSegmentsWithoutRouteParams(
+          currentNodePath,
+        );
+        const isMatching = parsedPath.reduce(
+          (acc: boolean, segment: string) => acc && pathname.includes(segment),
+          true,
+        );
+        if (isMatching) {
+          selectSubMenu(currentNode, parent);
+          return;
+        } else {
+          selectedNode ? setSelectedNode(null) : setSavedNode(null);
+        }
       }
     }
 
-    const universe = findUniverse(currentNavigationNode, pathname);
-    if (universe) {
-      selectSubMenu(universe.node, universe.parent);
+    const foundNode = findNodeByRouting(currentNavigationNode, pathname);
+    if (foundNode) {
+      selectSubMenu(foundNode.node, foundNode.universe);
     }
-  }, [currentNavigationNode]);
+  }, [currentNavigationNode, location]);
```

Comment:
Can you add some comments on this, as it is quite hard to understand what we are trying to achieve